### ISSUE TITLE
SkillLevelingManager Investigation

### ISSUE THREAD

**Issue by garrettluskey** [araex/BannerlordCoop#164]
Investigate the SkillLevelingManager class and determine what kind of synchronization will be needed if any.

**Comment by garrettluskey** [comment]
MobileParty.MainParty is always the player that is controlled in this case.

**Comment by Allen-Glass** [comment]
TL;DR: not worth syncing anything in this class. 

From looking into this, here's what I am seeing... There are no properties on SkillLevelingManager. This class can be seen more as a static helper class that does calculations and maps xp totals to skills. Within this class there are a large number of public methods that do the basic math to calculate what your progression is whenever you perform an action (ex. selling an item for profit skillXp calculation). All these public methods end up calling one of the 4 private methods and each of these private methods eventually end up calling the AddSkillXp method within HeroDeveloper that sets the SkillObject with the skillXp.


Here's a simplified call stack of what happens when a static method within SkillLevelingManager is called (assuming action is taken from MobileParty.MainParty). 

- SkillLevelingManager.OnBribeGiven(int amount)
- SkillLevelingManager.OnPartySkillExercised(MobileParty party, SkillObject skill, float skillXp, SkillEffect.PerkRole perkRole = SkillEffect.PerkRole.None)
- Hero.AddSkillXp(SkillObject skill, float xpAmount)
- HeroDeveloper.AddSkillXp(SkillObject skill, float rawXp, bool isAffectedByFocusFactor = true, bool shouldNotify = true) //note: when hero calls this, both default bools are set as true

**Comment by garrettluskey** [comment]
#273 will handle all skills.
